File: isma-hsm/src/main/java/ru/nstu/isma/core/hsm/linear/LinearSystemMatrixBuilder.java
package ru.nstu.isma.core.hsm.linear;

import common.ClassBuilder;
import common.JavaClassBuilder;
import ru.nstu.isma.core.hsm.HSM;
import common.IndexMapper;

/**
 * Created by Bessonov Alex
 * on 14.03.2015.
 */
public class LinearSystemMatrixBuilder extends ClassBuilder<LinearSystemMatrix> {
    private final IndexMapper modelContext;
    public LinearSystemMatrixBuilder(IndexMapper modelContext) {
        this.modelContext = modelContext;
    }

    public LinearSystemMatrix build(String name, boolean printJava) {
        return super.build(name, "ru.nstu.isma.core.hsm.linear.", printJava);
    }

    @Override
    public String getJavaString(String name) {
        JavaClassBuilder javaClassBuilder = new JavaClassBuilder(modelContext);
        return javaClassBuilder.printLinearSystemMatrix(name);
    }
}

File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/linear/LinearSystemMatrixBuilder.kt
package ru.nstu.isma.hsm.linear

import ru.nstu.isma.hsm.HSM
import ru.nstu.isma.hsm.common.ClassBuilder
import ru.nstu.isma.hsm.common.IndexMapper
import ru.nstu.isma.hsm.common.JavaClassBuilder

/**
 * Created by Bessonov Alex
 * on 14.03.2015.
 */
class LinearSystemMatrixBuilder : ClassBuilder<LinearSystemMatrix?> {
    constructor(modelContext: IndexMapper?) : super(modelContext) {}
    constructor(hsm: HSM) : super(hsm) {}

    fun build(name: String?, printJava: Boolean): LinearSystemMatrix? {
        return super.build(name!!, "ru.nstu.isma15.hsm.linear.", printJava)
    }

    override fun getJavaString(name: String): String? {
        val javaClassBuilder = JavaClassBuilder(modelContext!!)
        return javaClassBuilder.printLinearSystemMatrix(name)
    }
}